src/api/goods.ts

```diff
@@ -1,11 +1,27 @@
 // eslint-disable-next-line
 const API_URL = `https://mate-academy.github.io/react_dynamic-list-of-goods/goods.json`;
 
-export function getAll(): Promise<Good[]> {
+export const getAll = (): Promise<Good[]> => {
   return fetch(API_URL)
-    .then(response => response.json());
-}
+    .then((response) => {
+      if (!response.ok) {
+        throw new Error(`${response.status} - ${response.statusText}`);
+      }
 
-export const get5First = () => {};
+      return response.json();
+    });
+};
 
-export const getRedGoods = () => {};
+export const get5First = async () => {
```

Comment:
useCallback

Reply:
if I am using useCallBack here it gives error that
React Hook "useCallback" cannot be called at the top level. React Hooks must be called in a React function component or a custom React Hook function.

Reply:
:harold: